### DISCUSSION TITLE
How is Supabase data handled?

### DISCUSSION THREAD

**Discussion by churichard** [supabase/supabase#3038] [Questions] [answered]
Where is Supabase data (e.g. in the database and in storage) stored? Is it encrypted at rest, either by Supabase or by the storage provider?

**Comment by darora** [comment, marked as answer]
Hey @churichard - for our hosted offering, we currently use various storage mechanisms offered by AWS, including S3 and EBS, though the details are likely to change and evolve over time. However, as an invariant, we do encrypt at rest all projects, including databases, and their associated storage objects.

**Reply by churichard** [reply]
Awesome, thanks for answering!

**Reply by AntoineRobert** [reply]
Great to hear that ! :100:

**Reply by samcfo** [reply]
Thanks a lot for this, our product is aimed at professionals, some of which require encryption of data at rest!

**Reply by sixcorners** [reply]
@darora does this mean that it's possible to lose all of our data if the key is lost? How is the key stored?

**Reply by inian** [reply]
Some of the encryption is transparently handled by AWS and some of it is managed by our own keys stored in [KMS](https://aws.amazon.com/kms/).

**Reply by realvandi** [reply]
A little bit late to the party, but appreciate the answer! 😄 Is there a specific FAQ/Help/Documentation page at this point surrounding this topic?
